store/src/main/java/org/apache/rocketmq/store/dledger/DLedgerCommitLog.java

```diff
@@ -493,7 +497,99 @@ public PutMessageResult putMessage(final MessageExtBrokerInner msg) {
+        long eclipseTimeInLock;
+        long queueOffset = 0;
+        String key ="";
+        try {
+            beginTimeInDledgerLock =  this.defaultMessageStore.getSystemClock().now();
+            encodeResult = this.messageSerializer.serialize(messageExtBatch);
+            if(encodeResult.size()>0){
+                key=encodeResult.get(0).queueOffsetKey;
+                queueOffset = topicQueueTable.get(key);
+            }else{
+                return new PutMessageResult(PutMessageStatus.MESSAGE_ILLEGAL, null);
+            }
+
+            int totalMsgLen = 0;
+            int msgNum = 0;
+            msgIdBuilder.setLength(0);
+            long beginQueueOffset = queueOffset;
+            long wroteOffset=0;
+
+            for (int i = 0; i <encodeResult.size() ; i++) {
+                long msgLen= encodeResult.get(i).data.length;
+                if (encodeResult.get(i).status  != AppendMessageStatus.PUT_OK) {
+                    log.warn("Put messages the message is illegal");
+                    return new PutMessageResult(PutMessageStatus.MESSAGE_ILLEGAL, new AppendMessageResult(encodeResult.get(i).status));
+                }
+                AppendEntryRequest request = new AppendEntryRequest();
+                request.setGroup(dLedgerConfig.getGroup());
+                request.setRemoteId(dLedgerServer.getMemberState().getSelfId());
+                request.setBody(encodeResult.get(i).data);
+                dledgerFuture = (AppendFuture<AppendEntryResponse>) dLedgerServer.handleAppend(request);
+                if (dledgerFuture.getPos() == -1) {
+                    log.warn("Put messages the os page cache is busy!");
+                    return new PutMessageResult(PutMessageStatus.OS_PAGECACHE_BUSY, new AppendMessageResult(AppendMessageStatus.UNKNOWN_ERROR));
+                }
+                wroteOffset =  dledgerFuture.getPos() + DLedgerEntry.BODY_OFFSET;
+                ByteBuffer buffer = ByteBuffer.allocate(MessageDecoder.MSG_ID_LENGTH);
+                String msgId = MessageDecoder.createMessageId(buffer, messageExtBatch.getStoreHostBytes(), wroteOffset);
+                if (msgIdBuilder.length() > 0) {
+                    msgIdBuilder.append(',').append(msgId);
+                } else {
+                    msgIdBuilder.append(msgId);
+                }
+                msgNum++;
+                queueOffset++;
+                totalMsgLen +=msgLen;
+				DLedgerCommitLog.this.topicQueueTable.put(key, queueOffset);
+            }
+            eclipseTimeInLock = this.defaultMessageStore.getSystemClock().now() - beginTimeInDledgerLock;
+            appendResult=new AppendMessageResult(AppendMessageStatus.PUT_OK, wroteOffset, totalMsgLen, msgIdBuilder.toString(),
+                    messageExtBatch.getStoreTimestamp(), beginQueueOffset, DLedgerCommitLog.this.defaultMessageStore.now() - beginTimeInDledgerLock);
+            appendResult.setMsgNum(msgNum);
```

Comment:
You should call dledgerFuture.get() to get AppendEntryResponse, but I didn't see any related code.